zookeeper-server/src/main/java/org/apache/zookeeper/util/SecurityUtils.java

```diff
@@ -254,15 +264,67 @@ public SaslServer run() {
         return null;
     }
 
+    /**
+     * get specific section from configuration or thrown
+     * an IOException when no section exists in configuration
+     *
+     * @param configuration Configuration instance
+     * @param section section name
+     * @return AppConfigurationEntry list
+     */
+    public static AppConfigurationEntry[] getAppConfigurationEntryWithSection(Configuration configuration, String section) throws IOException {
+        AppConfigurationEntry[] configurationEntries = configuration.getAppConfigurationEntry(section);
+        if (configurationEntries == null) {
+            String errorMessage = "Could not find a '" + section + "' entry in this configuration: Server cannot start.";
+            LOG.error(errorMessage);
+            throw new IOException(errorMessage);
+        }
+        return configurationEntries;
+    }
+
+
+     /**
+     * make server credentials map from configuration's server section.
+     * @param configuration Configuration instance
+     * @return Server credentials map
+     */
+    public static Map<String, String> getServerCredentials(Configuration configuration) throws IOException {
+        AppConfigurationEntry[] serverAppConfigurationEntry = getServerAppConfigurationEntry(configuration);
+        Map<String, String> credentials = new HashMap<>();
+        for (AppConfigurationEntry entry : serverAppConfigurationEntry) {
+            Map<String, ?> options = entry.getOptions();
+            // Populate DIGEST-MD5 user -> password map with JAAS configuration entries from the "Server" section.
```

Comment:
backlog: I suspect we should do similar to ZOOKEEPER-4753(https://github.com/apache/zookeeper/commit/e2070bed85d8b0c98a5a0045bf92421f473c412e#diff-04a0c58edb7731a3998fcaa021b8e2190d8c3947dfc87606bfe8f5a0e9423434R71) here, but this should be a separate issue.